gammapy/datasets/utils.py

```diff
@@ -203,3 +214,165 @@ def split_dataset(dataset, width, margin, split_template_models=True):
                 d.models = dataset.models
             datasets.append(d)
     return datasets
+
+
+def create_map_dataset_from_dl4_irfs(obs, geom=None, energy_axis_true=None, name=None):
+    """Create a map dataset from an observation containing DL4 IRFs
+
+
+    Parameters
+    ----------
+    obs : `~gammapy.data.Observation`
+        Observation containing DL4 IRFs
+    geom : `~gammapy.maps.WcsGeom`, optional
+        Output dataset maps geometry. The default is None, and it is derived from IRFS
+    energy_axis_true : `~gammapy.maps.MapAxis`, optional
+        True energy axis used for IRF maps. The default is None, and it is derived from IRFS
+    name : str, optional
+        Dataset name. The default is None, and the name is randomly generated.
+
+    Returns
+    -------
+    dataset : `~gammapy.datasets.MapDataset`
+        Map dataset.
+
+    """
+    from gammapy.makers import MapDatasetMaker
+    from gammapy.datasets import MapDataset
+
+    # define target geom
+    if geom is None:
+        geom_image = obs.aeff.geom.to_image()
+        geom = geom_image.to_cube([obs.edisp.edisp_map.geom.axes["energy"]])
+
+    energy_axis = geom.axes["energy"]
+
+    if energy_axis_true is None:
+        # energy_axis_true = obs.edisp.edisp_map.geom.axes["energy_true"]
+        energy_axis_true = obs.aeff.geom.axes["energy_true"]
+
+    # ensure that DL4 IRFs have the axes
```

Comment:
Why do DL4 irfs need the same energy_true?

Reply:
I think in any case exposure and edsip must have the same energy_true for the evaluator to work.

Reply:
Yes, but this should be guaranteed for fermipy as well. Or does it output inconsistent energy axis definitions?

Reply:
As far as I understand by default fermipy `compute_drm` use `edisp_bins=1` and not the value set in the config see [here](https://fermipy.readthedocs.io/en/latest/_modules/fermipy/gtanalysis.html#GTAnalysis.compute_drm).
Moreover from what I tested using negative edisp_bins<0 in `compute_drm` also add extra bins to the drm but that's not the case for the exposure. So only `edisp_bins>0` is compatible with what is expected by gammapy.

Reply:
I see, but is the binning still aligned? So a simple slicing would remove the additional bin and make it consistent?

Reply:
no they are shifted by half a bin, exposure centres are edisp edges (same as counts but with `edisp_bins` more in true energy)